### ISSUE TITLE
NVDA 2023.2 RC1 not reading Google Docs

### ISSUE THREAD

**Issue by andrew-l-d** [nvaccess/nvda#15353]
### Steps to reproduce:
1.  Open a Google Doc.  For example:
https://docs.google.com/document/d/1rnLsKf_tCk_M9XIjT7ecrGH30zFWh7vLKzux-Inne-c/edit?usp=sharinglp1/?gclid=CjwKCAiAwrf-BRA9EiwAUWwKXsKKt04AiVNaH0BkDdJXzoZUBG7icgaGtEi-kmUd9KLEm_4o0hMoZxoCtZcQAvD_BwE
2. Press alt+shift+t to open the Tools Menu, arrow up to Accessibility and ensure that Screen Reader Access is checked
3. Ensure that NVDA is in focus mode and try to read the contents.
### Actual behavior:

tool bar
Chrome  menu button  subMenu  Customize and control Google Chrome
down arrow
menu  selected  subMenu
escape
tool bar
Chrome  menu button  subMenu  Customize and control Google Chrome
escape
application
document
blank
down arrow
blank
down arrow
blank
NVDA+space
down arrow
up arrow
down arrow
down arrow
alt+tab
page.txt - Notepad  row 1  column 2
alt+tab
maths-article.html - Notepad++  column 3
alt+tab
New Issue · nvaccess/nvda — Mozilla Firefox  column 4
alt+tab
C:\Users\Andrew\Documents\util\nvda-files  row 2  column 1
alt+tab
Inbox - access_tech@iprimus.com.au - Outlook  column 2
alt+tab
C:\Users\Andrew\OneDrive\DOCUMENTS  column 3
alt+tab
NoteTab Pro (Acc)  -  ozewai-committee.txt  column 4
alt+tab
Spotify Premium  row 3  column 1
alt+tab
Chat | Microsoft Teams  column 2
alt+tab
NVDA Speech Viewer  column 3
NVDA Speech Viewer
Show Speech Viewer on Startup  check box  not checked  Alt+s
down arrow
Nothing is spoken.
### Expected behavior:

Pressing down arrow should result in each line of text being presented.
### NVDA logs, crash dumps and other attachments:
nvda-ad.log attached

#### NVDA installed/portable/running from source:
Installed and temporary
#### NVDA version:
2023.2 RC1
#### Windows version:
10
#### Name and version of other software in use when reproducing the issue:
Google Chrome or Mozilla Firefox
#### Other information about your system:
All updates done.  Cannot think of anything else significant.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA up to 2023.1 behave normally
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
[NVDA-AD.LOG](https://github.com/nvaccess/nvda/files/12485630/NVDA-AD.LOG)
[NVDA-AD.LOG](https://github.com/nvaccess/nvda/files/12485636/NVDA-AD.LOG)
[NVDA-AD.LOG](https://github.com/nvaccess/nvda/files/12485647/NVDA-AD.LOG)

**Comment by Brian1Gaff** [comment]
Did you do a restart of the machine as well?
 To be honest, Google docs is a bit of a mess at the best of times, and I now always ask anyone sending a file to send me word odt oreven text rather than google.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by andrew-l-d** [comment]
Not only have I restarted the machine, I have created a temporary copy of 2023.2 RC1 and left everything at pretty close to default.  If I switch between 2023.1 and 2023.2 RC1 while leaving the browser open the file reads and does not read respectively.  Will try on another machine soonish.  Have not checked with Sheets yet.

Andrew

**Comment by andrew-l-d** [comment]
All fixed!  I tried my second machine and it worked correctly.  Devoid of other ideas, I cleared user data in Chrome on the first machine and now it works.
Apologies for any alarm caused.  I will close the ticket.
Andrew

**Comment by andrew-l-d** [comment]
Sadly, my celebration was unfounded.  Google Docs have stopped reading completely on my main machine again with NVDA 2023.2 RC1 and there is partial access with 2023.1.  My notebook just did an update and issues are similar there.  I cannot tell if it is a browser or OS issue, but there is definitely a problem reading Google Docs.  Both machines are running Windows 10.  I can no longer read a Google Doc in browse mode and capital letters are not announced as capitals.

Andrew

**Comment by andrew-l-d** [comment]
I seem to have fixed the issue.  I am fairly certain that the problem was caused my Google Chrome inviting me to increase security settings.  I seem to have reverted, but am having trouble getting back to the relevant settings.  The problem started on both machines following an upgrade, so others may come across it if needing to use Google Docs.  I will close this ticket again.
Andrew

**Comment by nibarAhmed** [comment]
i have this issue as well running NVDA 2023.2. but it seems that this issue also occurs when using jaws.

**Comment by seanbudd** [comment]
@nibarAhmed - have you gone through the troubleshooting steps outlined by @andrew-l-d in this issue? This includes clearing your browser cache, restarting your device and enabling accessibility mode in google docs. If you are still experience this can you please fill out a new issue?

**Comment by andrew-l-d** [comment]
Following considerable investigation I can report the following:
NVDA 2023.2 is behaving exactly the same as previous versions when reading Google docs.  The reason for my confusion is that some Google docs result differently when Braille Support is enabled.
When reading Google docs I prefer to check both Screen Reader Support and Braille Support (no Braille display connected).  That allows me to switch to browse mode, which is convenient in some circumstances (with Braille Support unchecked browse mode is not available).
In some, but not all, documents when using Chrome with Braille Support enabled NVDA will not read the contents.  I have not yet been able to discover why this is but will discuss with the document's author.  The problem does not occur with files I create.  I can also copy the contents of a troublesome file and paste it into one I have created and I can use browse mode.
I can open the troublesome files using Firefox with Braille Support enabled and can access the file.  Browse mode is also available.
In summary:  If wanting to use browse mode it is necessary to turn on Braille Support.  This is effective in all files I have encountered so far when using Firefox.
Apologies for the false alarms.

Andrew

**Comment by Leonardo-pena** [comment]
@andrew-l-d 
This is not a false alarm. this was truely helpful. I was having this issue for the longest, and the only solution someone gave me was just to enable braille support and screen reader accessibility support. I do find it interesting that screen reader accessibility is necessary for braille support to be enabled, so I wonder if they are both linked and require both to be checked in order to work. 
In any case, will definitely make sure to share this. 
Also, to enable braille support while in google docs, just hit control+alt+ h, or command+options+h.
Also, while in the doc, if I toggle braille support, it automatically causes screen reader support to stop working for NVDA. 
Hope that helps
Keep safe, and shalom. 
Leo Pena

**Comment by nibarAhmed** [comment]
After further investigation I believe  there is a setting that is making google docx not read. I believe this is tied to a google account option since when using certain google accounts documents are not readable but when switching to another chrome profile running on other google accounts it works. Something  to note is that for google accounts where documents are not being read it seems to work as long as you are the owner. but when viewing a shared document the document is not read at all. this is with both screen reader support and braile support enabled. Seems to be a google issue since I can reproduce the same behaviour using jaws.
